proposals/p5270.md

```diff
@@ -0,0 +1,132 @@
+Exceptions. See /LICENSE for license information.
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+-->
+
+[Pull request](https://github.com/carbon-language/carbon-lang/pull/5270)
+
+<!-- toc -->
+
+## Table of contents
+
+-   [Abstract](#abstract)
+-   [Problem](#problem)
+-   [Background](#background)
+-   [Proposal](#proposal)
+-   [Rationale](#rationale)
+-   [Alternatives considered](#alternatives-considered)
+    -   [Use a branch to refer to the explorer codebase](#use-a-branch-to-refer-to-the-explorer-codebase)
+    -   [Do nothing](#do-nothing)
+
+<!-- tocstop -->
+
+## Abstract
+
+The explorer codebase is no longer actively developed or maintained, but retains
+value as a place to look for implementations of various parts of the design.
+Having both the explorer and toolchain in the working git repository causes
+confusion for developers and users, and increases our incremental maintenance
+burden. We will move explorer out of the working git repository so that it
+remains accessible but is clearly deliniated from the toolchain and is more
+clearly frozen or archived.
+
+## Problem
+
+Users have expressed confusion when they come across examples or tests for the
+explorer that involve `Main()`, whereas the design and implementation of the
+toolchain have moved on to `Run()` as the entry point.
+
+Developers have attempted to update the explorer codebase to fix style guide
+issues[^1], as tools like grep or clang-tidy do not differentiate our active
+toolchain codebase from the frozen explorer one.
+
+As we update Bazel, Clang, Clang-tidy, etc, we also have to update the explorer
+codebase to keep it building cleanly.
+
+As we modify the `//testing` harness, we have to accommodate and work around the
+explorer tests[^2][^3][^4].
+
+[^1]: https://github.com/carbon-language/carbon-lang/pull/5224
+
+[^2]: https://github.com/carbon-language/carbon-lang/pull/4979
+
+[^3]: https://github.com/carbon-language/carbon-lang/pull/5025
+
+[^4]: https://github.com/carbon-language/carbon-lang/pull/5036
+
+## Background
+
+-   Previous proposal:
+    [p3532: Focus implementation effort on the toolchain](https://github.com/carbon-language/carbon-lang/blob/trunk/proposals/p3532.md)
+-   Kick-off of this discussion
+    [#5224](https://github.com/carbon-language/carbon-lang/pull/5224#pullrequestreview-2730512195)
+-   Discord discussion:
+    [2025-04-02 #explorer](https://discord.com/channels/655572317891461132/763516049710120960/1356979197070803095)
+
+## Proposal
+
+1. Add a tag `explorer-archived` in the main `carbon-lang` git repository.
+2. Copy the `//explorer` directory and its dependencies to a new `explorer`
+   repository under the `carbon-language` organization.
+3. Locally ensure the explorer tests build and pass under the `explorer`
+   repository.
+4. Add a `README.md` to the `explorer` repository that explains explorer is
+   archived and not under active development.
+5. Delete `//explorer` in the main `carbon-lang` repository.
```

Comment:
Is Compiler Explorer still building explorer (visible at https://carbon.compiler-explorer.com/z/b61T6n44h)? Maybe something to double-check if you're not sure -- even with the archive repo, might want to just ensure it's turned off.

Reply:
I am unsure if it's building it still. There's nothing visible in the compiler-explorer git repo that would suggest either way. But since the binary isn't part of the nightly release tarball, and we don't provide a binary of the explorer in some other way, I am guessing that it is still being built.

If this is approved I will kick off a PR with compiler-explorer to remove the carbon-explorer, and that can prompt them to also stop building it.

I've added this as a step.

Reply:
FWIW, I suspect Chandler's ideal result would be leaving it on Compiler Explorer, just not building new versions.

Reply:
No strong opinion TBH. Whatever is easiest.

Reply:
Hi folks, we're fine with keeping the existing explorer versions. Plus it means old links won't break even though the upstream explorer is gone.